crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs

```diff
@@ -0,0 +1,400 @@
+use oxc_ast::{
+    ast::{Expression, MemberExpression},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn prefer_object_has_own_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "Disallow use of `Object.prototype.hasOwnProperty.call()` and prefer use of `Object.hasOwn()`."
+    ).with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferObjectHasOwn;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow use of `Object.prototype.hasOwnProperty.call()` and prefer use of `Object.hasOwn()`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It is very common to write code like:
+    /// ```javascript
+    /// if (Object.prototype.hasOwnProperty.call(object, "foo")) {
+    ///     console.log("has property foo");
+    /// }
+    /// ```
+    /// This is a common practice because methods on Object.prototype can sometimes be unavailable or redefined (see the no-prototype-builtins rule).
+    /// Introduced in ES2022, Object.hasOwn() is a shorter alternative to Object.prototype.hasOwnProperty.call():
+    /// ```javascript
+    /// if (Object.hasOwn(object, "foo")) {
+    ///   console.log("has property foo")
+    /// }
+    /// ```
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// Object.prototype.hasOwnProperty.call(obj, "a");
+    /// Object.hasOwnProperty.call(obj, "a");
+    /// ({}).hasOwnProperty.call(obj, "a");
+    /// const hasProperty = Object.prototype.hasOwnProperty.call(object, property);
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// Object.hasOwn(obj, "a");
+    /// const hasProperty = Object.hasOwn(object, property);
+    /// ```
+    PreferObjectHasOwn,
+    style,
+    conditional_fix
+);
+
+impl Rule for PreferObjectHasOwn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+
+        let Some(callee) = unwrap_parenthesized_expr(&call_expr.callee).get_member_expr() else {
```

Comment:
I think you want `get_inner_expression`

Reply:
thanks!
fixed af29d35e83c269a695965c17f4e4063d265168ff